test/rules/exhaustive_cases_test.dart

```diff
@@ -15,160 +40,223 @@ class E {
-  Subclass() : super._();
+  static const e = E._(1);
+  static const f = E._(2);
+  static const g = E._(3);
 }
 
-void s(Subclassed e) {
-  switch (e) { // OK
-    case Subclassed.e:
+void okParenthesized(E e) {
+  switch (e) {
+    case (E.e):
+      print('e');
+      break;
+    case ((E.f)):
       print('e');
+      break;
+    case (E.g):
+      print('e');
+      break;
   }
 }
+''');
+  }
 
-// Not Enum-like.
+  test_notEnumLike_ok() async {
+    await assertNoDiagnostics(r'''
 class TooFew {
   const TooFew._();
 
   static const e = TooFew._();
 }
 
 void t(TooFew e) {
-  switch (e) { // OK
+  switch (e) { 
     case TooFew.e:
       print('e');
   }
 }
 
-// Not Enum-like.
 class PublicCons {
   const PublicCons();
   static const e = PublicCons();
   static const f = PublicCons();
 }
 
 void p(PublicCons e) {
-  switch (e) { // OK
+  switch (e) { 
     case PublicCons.e:
       print('e');
   }
 }
+''');
+  }
 
-// Handled by analyzer.
-enum ActualEnum { e, f }
+  test_notEnumLike_subclassed_ok() async {
+    await assertNoDiagnostics(r'''
+class Subclassed {
+  const Subclassed._();
 
-void ae(ActualEnum e) {
-  // ignore: missing_enum_constant_in_switch
-  switch (e) { // OK
-    case ActualEnum.e:
-      print('e');
-  }
+  static const e = Subclassed._();
+  static const f = Subclassed._();
+  static const g = Subclassed._();
 }
 
-// Some fields are deprecated.
-class DeprecatedFields {
-  final int i;
-  const DeprecatedFields._(this.i);
-
-  @deprecated
-  static const oldFoo = newFoo;
-  static const newFoo = DeprecatedFields._(1);
-  static const bar = DeprecatedFields._(2);
-  static const baz = DeprecatedFields._(3);
+class Subclass extends Subclassed {
+  Subclass() : super._();
 }
 
-void dep(DeprecatedFields e) {
-  switch (e) { // OK
-    case DeprecatedFields.newFoo:
-      print('newFoo');
-      break;
-    case DeprecatedFields.bar:
-      print('bar');
-      break;
-    case DeprecatedFields.baz:
-      print('baz');
-      break;
+void s(Subclassed e) {
+  switch (e) { 
+    case Subclassed.e:
+      print('e');
+  }
+}
+''');
   }
+}
 
-  switch (e) { // LINT
-    case DeprecatedFields.newFoo:
-      print('newFoo');
-      break;
-    case DeprecatedFields.baz:
-      print('baz');
-      break;
+@reflectiveTest
+class ExhaustiveCasesTestLanguage219 extends BaseExhaustiveCasesTest
+    with LanguageVersion219Mixin {
+  test_enum_ok() async {
+    await assertDiagnostics(actualEnumSource, [
+      error(StaticWarningCode.MISSING_ENUM_CONSTANT_IN_SWITCH, 52, 10),
+    ]);
   }
+}
 
-  switch (e) { // OK
-    case DeprecatedFields.oldFoo:
-      print('oldFoo');
-      break;
-    case DeprecatedFields.bar:
-      print('bar');
-      break;
-    case DeprecatedFields.baz:
-      print('baz');
-      break;
+@reflectiveTest
+class ExhaustiveCasesTestLanguage300 extends BaseExhaustiveCasesTest
+    with LanguageVersion300Mixin {
+  @override
+  List<String> get experiments => ['records', 'patterns'];
```

Comment:
Can we put these experiments into `LanguageVersion300Mixin`?

Reply:
Good idea!